mathics/builtin/image.py

```diff
@@ -1085,38 +1608,84 @@ def apply(self, image, stype, evaluation):
         pixels = image.pixels
         stype = stype.get_string_value()
         if stype == 'Real':
-            pixels = skimage.img_as_float(pixels)
+            pixels = pixels_as_float(pixels)
         elif stype == 'Byte':
-            pixels = skimage.img_as_ubyte(pixels)
+            pixels = pixels_as_ubyte(pixels)
         elif stype == 'Bit16':
-            pixels = skimage.img_as_uint(pixels)
+            pixels = pixels_as_uint(pixels)
         elif stype == 'Bit':
-            pixels = pixels.as_dtype(numpy.bool)
+            pixels = pixels.astype(numpy.bool)
         else:
             return evaluation.message('ImageData', 'pixelfmt', stype)
         return from_python(pixels.tolist())
 
 
 class ImageTake(_ImageBuiltin):
+    '''
+    <dl>
+    <dt>'ImageTake[$image$, $n$]'
+      <dd>gives the first $n$ rows of $image$.
+    <dt>'ImageTake[$image$, -$n$]'
+      <dd>gives the last $n$ rows of $image$.
+    <dt>'ImageTake[$image$, {$r1$, $r2$}]'
+      <dd>gives rows $r1$, ..., $r2$ of $image$.
+    <dt>'ImageTake[$image$, {$r1$, $r2$}, {$c1$, $c2$}]'
+      <dd>gives a cropped version of $image$.
+    </dl>
+    '''
+
     def apply(self, image, n, evaluation):
         'ImageTake[image_Image, n_Integer]'
-        return Image(image.pixels[:int(n.to_python())], image.color_space)
+        py_n = n.get_int_value()
+        if py_n >= 0:
+            pixels = image.pixels[:py_n]
+        elif py_n < 0:
+            pixels = image.pixels[py_n:]
+        return Image(pixels, image.color_space)
+
+    def apply_rows(self, image, r1, r2, evaluation):
+        'ImageTake[image_Image, {r1_Integer, r2_Integer}]'
+        py_r1 = max(r1.get_int_value() - 1, 0)
+        py_r2 = max(r2.get_int_value() - 1, 0)
+        return Image(image.pixels[py_r1:1 + py_r2], image.color_space)
```

Comment:
also need to check the indices are valid `0 <= py_r1 <= py_r2 < len(image.pixels)`. At the moment it looks like we're just checking `py_ri >= 0`.